From 75dfffdf6ab5f0f7b76fc6fc2605568b7b092b05 Mon Sep 17 00:00:00 2001
From: Neil Horman <nhorman@redhat.com>
Date: Wed, 18 Jun 2014 16:20:16 -0400
Subject: [redhat] docs: Modify RHELPatchRules to loosen requirements regarding unsupported code

Message-id: <1403108416-5565-1-git-send-email-nhorman@redhat.com>
Patchwork-id: 84261
O-Subject: [RHEL7 PATCH] redhat: Modify RHELPatchRules to loosen requirements regarding unsupported code
Bugzilla:
RH-Acked-by: John Greene <jogreene@redhat.com>
RH-Acked-by: Mike Snitzer <snitzer@redhat.com>
RH-Acked-by: Jesper Brouer <jbrouer@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Veaceslav Falico <vfalico@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>

About a year or so ago, I tried to implement this change, and while there was
general technical support for it, there was some external-to-the-list
disagreement about allowing the patching of unsupported code.  This has been a
frustration to me, in that despite repeated questions, we've never gotten a
clear answer on what this is the stance.  Given prarits recent patch that is
useful, but seems to be blocked on the fact that we can't take it because it
affects unsupported code, I'd like to re-propose this, in the hopes of actually
making the change or forcing the conversation so we can all understand why that
is the case.

In short, this patch changes our rule regarding the patching of unsupported
code.  It alters the language to address the real concern, which is that we
should not allow the enabling of unsupported code, while allowing us to actually
change unsupported code, in an effort to ease the burden of having to hack up
upstream cherry-picks, which wastes time during backports

Signed-off-by: Neil Horman <nhorman@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/redhat/docs/RHELPatchRules b/redhat/docs/RHELPatchRules
index 6c42bbc..95dc958 100644
--- a/redhat/docs/RHELPatchRules
+++ b/redhat/docs/RHELPatchRules
@@ -403,13 +403,12 @@ Some of the common approaches to generating a patch are:
 build.  If this is not the case, the patch will be rejected by reviewers and
 maintainers.
 
-[Required] Do not patch unsupported code.
+[Required] Do not enable unsupported functionality
 
-    Currently, we only patch drivers, subsystems, and architectures that we
-actively support.  The reason for this is that in the past, patches to
-unsupported portions of the kernel have resulted in support requests for
-components we do not wish to support.  Even a single support request of this
-type can lead to significant issues for Partner and Product Management.
+    In order to avoid customers requesting support for features that Red Hat is
+not prepared to support, code changes must not inadvertently enable features or
+functionality unless expressly stated in the approved bugzilla for which the
+code change is being made.
 
     Postings of large firmware patches will be rejected by the list server.
 Instead of posting the full firmware patch, a "stub" patch with a valid patch
-- 
1.7.1